priv/script.sh

```diff
@@ -2,441 +2,489 @@
+    ln -sf "$phx_tools_fish_path" "$fish_confd"
+  fi
+}
 
-    printf "${white}\n"
-    sleep 1.5
-    maybe_install "Elixir"
+add_env() {
+  # Ask for sudo password upfront
+  cat <<SUDO
+phx.tools may need to install build dependencies with your platform package
+manager. We are requesting 'sudo' permission before starting for a smoother
+install.
+
+SUDO
+  sudo -v
+
+  # Keep sudo alive
+  while true; do
+    sudo -n true
+    sleep 60
+    kill -0 "$$" || exit
+  done 2>/dev/null &
+
+  if [ "$os_type" = macOS ]; then
+    maybe_install "Xcode Command Line Tools"
+    maybe_install Homebrew
+  else
+    maybe_install git
+  fi
 
-    printf "${white}\n"
-    sleep 1.5
-    maybe_install "Phoenix"
+  maybe_install "mise"
+  maybe_install "Erlang"
+  maybe_install "Elixir"
+  maybe_install "Phoenix"
+  maybe_install "PostgreSQL"
 
-    printf "${white}\n"
-    sleep 1.5
-    maybe_install "PostgreSQL"
+  install_shell_scripts
 
-    printf "${white}\n"
-    printf "${cyan}${bold}phx.tools setup is complete!\n"
-    printf "${cyan}${bold}Please restart the terminal and type in the following command:\n"
-    printf "${cyan}${bold}mix phx.new\n"
-    printf "${white}\n"
+  printf "\n%sphx.tools setup is complete!\n\n" "$normal$cyan$bold"
+  printf "%sPlease restart the terminal and type in the following command:\n" \
+    "$cyan$bold"
+  printf "%s    mix phx.new\n%s\n" "$cyan$bold" "$normal"
 }
 
-phx_tools="
+banner='
         ██████╗░██╗░░██╗██╗░░██╗  ████████╗░█████╗░░█████╗░██╗░░░░░░██████╗ 
         ██╔══██╗██║░░██║╚██╗██╔╝  ╚══██╔══╝██╔══██╗██╔══██╗██║░░░░░██╔════╝ 
         ██████╔╝███████║░╚███╔╝░  ░░░██║░░░██║░░██║██║░░██║██║░░░░░╚█████╗░ 
         ██╔═══╝░██╔══██║░██╔██╗░  ░░░██║░░░██║░░██║██║░░██║██║░░░░░░╚═══██╗ 
         ██║░░░░░██║░░██║██╔╝╚██╗  ░░░██║░░░╚█████╔╝╚█████╔╝███████╗██████╔╝ 
         ╚═╝░░░░░╚═╝░░╚═╝╚═╝░░╚═╝  ░░░╚═╝░░░░╚════╝░░╚════╝░╚══════╝╚═════╝░ 
-"
-by="
+
                             ██████╗░██╗░░░██╗
                             ██╔══██╗╚██╗░██╔╝
                             ██████╦╝░╚████╔╝░
                             ██╔══██╗░░╚██╔╝░░
                             ██████╦╝░░░██║░░░
                             ╚═════╝░░░░╚═╝░░░
-"
-optimum="
+
     ░█████╗░██████╗░████████╗██╗███╗░░░███╗██╗░░░██╗███╗░░░███╗██████╗░██╗░░██╗
     ██╔══██╗██╔══██╗╚══██╔══╝██║████╗░████║██║░░░██║████╗░████║██╔══██╗██║░░██║
     ██║░░██║██████╔╝░░░██║░░░██║██╔████╔██║██║░░░██║██╔████╔██║██████╦╝███████║
     ██║░░██║██╔═══╝░░░░██║░░░██║██║╚██╔╝██║██║░░░██║██║╚██╔╝██║██╔══██╗██╔══██║
     ╚█████╔╝██║░░░░░░░░██║░░░██║██║░╚═╝░██║╚██████╔╝██║░╚═╝░██║██████╦╝██║░░██║
     ░╚════╝░╚═╝░░░░░░░░╚═╝░░░╚═╝╚═╝░░░░░╚═╝░╚═════╝░╚═╝░░░░░╚═╝╚═════╝░╚═╝░░╚═╝
-"
-
-printf "%s\n" "$phx_tools"
-printf "%s\n" "$by"
-printf "%s\n" "$optimum"
-
-sleep 3
-
-printf "\n"
-
-printf "${bblue}${bold}Welcome to the phx.tools shell script for ${os_type}.\n"
+'
 
-sleep 3
+printf "%s\n" "$bblue$banner"
```

Comment:
*All* of the print statements have been condensed to reduce the whitespace. It was scrolling too much on my 56-line shell.